test/run_test.py

```diff
@@ -782,6 +782,11 @@ def print_to_file(s):
             if not continue_through_error:
                 print_to_file("Stopping at first consistent failure")
                 break
+            if current_failure == f"'{test_file}'":
```

Comment:
My understanding is that the issue only manifests when `keep-going` is used, either in PR or in trunk by default.  How does the fix work in that case?  Does `test_file` var here include the test class or test name so that there is enough information to know that the same thing fails?  I need to re-read the code here a bit :)